src/coreclr/jit/hwintrinsic.cpp

```diff
@@ -1137,10 +1137,79 @@ GenTree* Compiler::impHWIntrinsic(NamedIntrinsic        intrinsic,
             case 2:
                 op2 = getArgForHWIntrinsic(sigReader.GetOp2Type(), sigReader.op2ClsHnd);
                 op2 = addRangeCheckIfNeeded(intrinsic, op2, mustExpand, immLowerBound, immUpperBound);
-                op1 = getArgForHWIntrinsic(sigReader.GetOp1Type(), sigReader.op1ClsHnd);
 
-                retNode = isScalar ? gtNewScalarHWIntrinsicNode(retType, op1, op2, intrinsic)
-                                   : gtNewSimdHWIntrinsicNode(retType, op1, op2, intrinsic, simdBaseJitType, simdSize);
+#ifdef TARGET_ARM64
+                if ((intrinsic == NI_AdvSimd_VectorTableLookup) || (intrinsic == NI_AdvSimd_Arm64_VectorTableLookup))
+                {
+                    op1                       = impPopStack().val;
+                    ClassLayout* layout       = op1->GetLayout(this);
+                    unsigned     structSize   = layout->GetSize();
+                    unsigned     slotCount    = layout->GetSlotCount();
+                    var_types    typeOfLayout = layout->GetType();
+                    if (typeOfLayout == TYP_STRUCT)
+                    {
+                        unsigned fieldCount = info.compCompHnd->getClassNumInstanceFields(sigReader.op1ClsHnd);
+                        op1->AsLclVar()->SetMultiRegUse();
+                        GenTreeFieldList* fieldList = new (this, GT_FIELD_LIST) GenTreeFieldList();
```

Comment:
The field list is created... but not used?

Reply:
True and I forgot to remove the code. I could get the struct's fields promoted by setting `lvUsedInSIMDIntrinsic = false`.